File: src/test/java/com/indiepost/search/TagSearchTest.java
package com.indiepost.search;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.hibernate5.Hibernate5Module;
import com.indiepost.NewIndiepostApplication;
import com.indiepost.dto.post.PostSummaryDto;
import com.indiepost.service.PostService;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.context.web.WebAppConfiguration;

import javax.inject.Inject;

/**
 * Created by jake on 17. 3. 16.
 */
@RunWith(SpringRunner.class)
@SpringBootTest(classes = NewIndiepostApplication.class)
@WebAppConfiguration
public class TagSearchTest {

    @Inject
    private PostService postService;

    @Test
    public void tagSearchShouldReturnPostsCorrectly() throws JsonProcessingException {
        Page<PostSummaryDto> dtoList = postService.findByTagName("독립영화", PageRequest.of(0, 10));
        ObjectMapper objectMapper = new ObjectMapper();
        objectMapper.registerModule(new Hibernate5Module());
        System.out.println("\n\n*** Start serialize PostsByTagName ***\n\n");
        String result = objectMapper.configure(SerializationFeature.INDENT_OUTPUT, true)
                .writeValueAsString(dtoList);
        System.out.println("Length of results: " + (dtoList.getContent().size()) + " posts");
        System.out.println("Size of results: " + (result.getBytes().length / 1024.0) + " kb");
        System.out.println(result);
    }
}

File: src/test/kotlin/com/indiepost/search/TagSearchTest.kt
package com.indiepost.search

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import com.fasterxml.jackson.datatype.hibernate5.Hibernate5Module
import com.indiepost.IndiepostBackendApplication
import com.indiepost.service.PostService
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.data.domain.PageRequest
import org.springframework.test.context.junit.jupiter.SpringExtension
import org.springframework.test.context.web.WebAppConfiguration
import javax.inject.Inject

/**
 * Created by jake on 17. 3. 16.
 */
@ExtendWith(SpringExtension::class)
@SpringBootTest(classes = [IndiepostBackendApplication::class])
@WebAppConfiguration
class TagSearchTest {

    @Inject
    private lateinit var postService: PostService

    @Test
    @Throws(JsonProcessingException::class)
    fun tagSearchShouldReturnPostsCorrectly() {
        val dtoList = postService.findByTagName("독립영화", PageRequest.of(0, 10))
        val objectMapper = ObjectMapper()
        objectMapper.registerModule(Hibernate5Module())
        println("\n\n*** Start serialize PostsByTagName ***\n\n")
        val result = objectMapper.configure(SerializationFeature.INDENT_OUTPUT, true)
                .writeValueAsString(dtoList)
        println("Length of results: " + dtoList.content.size + " posts")
        println("Size of results: " + result.toByteArray().size / 1024.0 + " kb")
        println(result)
    }
}
